hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/UserDetailsFormatter.java

```diff
@@ -0,0 +1,175 @@
+     *         be <null>
+     */
+    public static String loadAndFormatCreatedBy(final BaseEntity baseEntity) {
+        if (baseEntity == null || baseEntity.getCreatedBy() == null) {
+            return StringUtils.EMPTY;
+        }
+
+        return loadAndFormatUsername(baseEntity.getCreatedBy());
+    }
+
+    /**
+     * Load user details by {@link BaseEntity#getLastModifiedBy()} and format
+     * the user name. Use
+     * {@link UserDetailsFormatter#loadAndFormatUsername(String)}
+     * 
+     * @param baseEntity
+     *            the entity
+     * @return the formatted 'last modefied by user name' (max 100 characters)
+     *         cannot be <null>
+     */
+    public static String loadAndFormatLastModifiedBy(final BaseEntity baseEntity) {
+        if (baseEntity == null || baseEntity.getLastModifiedBy() == null) {
+            return StringUtils.EMPTY;
+        }
+
+        return loadAndFormatUsername(baseEntity.getLastModifiedBy());
+    }
+
+    /**
+     * Load user details by the current session information and format the user
+     * name to max 12 characters. @see
+     * {@link UserDetailsFormatter#loadAndFormatUsername(String, int)}
+     * 
+     * @param baseEntity
+     *            the entity
+     * @return the formatted user name (max 12 characters) cannot be <null>
+     */
+    public static String loadAndFormatCurrentUsername() {
+        return loadAndFormatUsername(getCurrentUser().getUsername(), 6);
+    }
+
+    /**
+     * Load user details by the user name and format the user name. If the
+     * loaded {@link UserDetails} is not a instance of a {@link UserPrincipal},
+     * then just the {@link UserDetails#getUsername()} will return.
+     * 
+     * If first and last name available, they will combined. Otherwise the
+     * {@link UserPrincipal#getLoginname()} will formatted. The formatted name
+     * is reduced to 100 characters.
+     * 
+     * @param username
+     *            the user name
+     * @param expectedNameLength
+     *            the name size of each name part
+     * @return the formatted user name (max 2 * expectedNameLength characters)
+     *         cannot be <null>
+     */
+    public static String loadAndFormatUsername(final String username, final int expectedNameLength) {
+        final UserDetails userDetails = loadUserByUsername(username);
+        if (!(userDetails instanceof UserPrincipal)) {
+            return userDetails.getUsername();
+        }
+
+        final UserPrincipal userPrincipal = (UserPrincipal) userDetails;
+
+        final String trimmedFirstname = trimAndFormatDetail(userPrincipal.getFirstname(), expectedNameLength);
+        final String trimmedLastname = trimAndFormatDetail(userPrincipal.getLastname(), expectedNameLength);
+
+        if (StringUtils.isEmpty(trimmedFirstname) && StringUtils.isEmpty(trimmedLastname)) {
+            return StringUtils.substring(userPrincipal.getLoginname(), 0, 2 * expectedNameLength);
```

Comment:
I think that from the caller's point of view it is easier to say: "I have 20 characters of space left, please give me a suitable sized username". So it is easier to call, ready and understand. I do not care about the internals as long as there is no requirement that firstname and lastname have to be equal in size.